src/PhpPact/Consumer/Model/Pact.php

```diff
@@ -114,8 +114,14 @@ private function withSpecification(): self
             '3.0.0' => $this->ffi->PactSpecification_V3,
             '4.0.0' => $this->ffi->PactSpecification_V4,
         ];
-        $version = $supportedVersions[$this->config->getPactSpecificationVersion()] ?? $this->ffi->PactSpecification_Unknown;
-        $this->ffi->pactffi_with_specification($this->id, $version);
+        $version = $this->config->getPactSpecificationVersion();
+        if (isset($supportedVersions[$version])) {
+            $specification = $supportedVersions[$version];
+        } else {
+            trigger_error(sprintf("Specification version '%s' is unknown", $version), E_USER_WARNING);
```

Comment:
Or using `print` like `PactTestListener`?

 `print sprintf("Specification version '%s' is unknown", $version);`